src/structures.jl

```diff
@@ -55,17 +55,77 @@ end
 mutable struct TulipaConstraint
     indices::DataFrame
     table_name::String
+    num_rows::Int
+    constraint_names::Vector{Symbol}
     expressions::Dict{Symbol,Vector{JuMP.AffExpr}}
+    duals::Dict{Symbol,Vector{Float64}}
 
     function TulipaConstraint(connection, table_name::String)
         return new(
             DuckDB.query(connection, "SELECT * FROM $table_name") |> DataFrame,
             table_name,
+            only([ # only makes sure that a single value is returned
+                row.num_rows for
+                row in DuckDB.query(connection, "SELECT COUNT(*) AS num_rows FROM $table_name")
+            ]), # This loop is required to access the query resulted, because it's a lazy struct
+            Symbol[],
+            Dict(),
             Dict(),
         )
     end
 end
 
+"""
+    attach_constraint!(model, cons, name, container)
+
+Attach a constraint named `name` stored in `container`, and set `model[name] = container`.
+This checks that the `container` length matches the stored `indices` number of rows.
+"""
+function attach_constraint!(
+    model::JuMP.Model,
+    cons::TulipaConstraint,
+    name::Symbol,
+    container::Vector{<:JuMP.ConstraintRef},
+)
+    @assert length(container) == cons.num_rows
+    push!(cons.constraint_names, name)
+    model[name] = container
+    return nothing
+end
+
+function attach_constraint!(model::JuMP.Model, cons::TulipaConstraint, name::Symbol, container)
+    # This should be the empty case container = Any[]
+    @assert length(container) == 0
+    @assert cons.num_rows == 0
+    empty_container = JuMP.ConstraintRef{JuMP.Model,Missing,JuMP.ScalarShape}[]
+    model[name] = empty_container
+    return nothing
+end
+
+"""
+    attach_expression!(cons, name, container)
+    attach_expression!(model, cons, name, container)
+
+Attach a expression named `name` stored in `container`, and optionally set `model[name] = container`.
+This checks that the `container` length matches the stored `indices` number of rows.
+"""
+function attach_expression!(cons::TulipaConstraint, name::Symbol, container::Vector{JuMP.AffExpr})
+    @assert length(container) = cons.num_rows
+    cons.expressions[name] = container
+    return nothing
```

Comment:
This is our friend Codecov doing it stuff, correct? I mean, it is an apparent hit in the code, but the only version misses it for whatever reason. I don't care about this as a problem, but just to be sure, I get it right.

Reply:
I fixed this. The `attach_expression!` was indeed not being used. I commented out the `model` version until we actually use it, though